src/collections/config/sanitize.ts

```diff
@@ -23,6 +30,42 @@ const sanitizeCollection = (config: Config, collection: CollectionConfig): Sanit
     isMergeableObject: isPlainObject,
   });
 
+  if (sanitized.timestamps !== false) {
+    let hasUpdatedAt = null;
+    let hasCreatedAt = null;
+    let i = 0;
+    while (hasCreatedAt === null && hasUpdatedAt === null && i < sanitized.fields.length) {
```

Comment:
_Should_ this be a while loop? I feel like a for loop would make more sense.

It seems like you would want to loop over all fields to check for _both_ `createdAt` and `updatedAt`. In this case, the loop will break when one of them is found and will not keep checking for the other.

I also wonder if we should be setting these fields `require: true` here instead of [here](https://github.com/payloadcms/payload/pull/2583/files#diff-1ea0586c0f08570bb5ba4938494ccaa6cdf35a0975ed7a51cacf73f5ab8ffb4bR409).

Reply:
You're right, my while loop condition is wrong. I used a while loop to stop iterating fields once createdAt and updatedAt were taken care of. I'll fix the condition.

The fields can't be `required` for the admin UI or the backend because of validation. The only reason I used `required: true` in entityToJSONSchema is for type generation. In the type they are always defined.